tests/codegen_testcases/solidity/borsh_decoding_complex_types.sol

```diff
@@ -12,10 +12,10 @@ contract Testing {
 	    
         // CHECK: ty:uint32 %temp.10 = (builtin ArrayLength ((arg #0)))
 	    // CHECK: ty:uint32 %temp.12 = uint32 0
+	    // CHECK: ty:uint32 %temp.13 = (builtin ReadFromBuffer ((arg #0), uint32 0))
```

Comment:
The instruction is misplaced because you are not validating the offset to read the `string[]` length.